packaging/Dockerfile.centos.7

```diff
@@ -17,7 +17,7 @@ ARG UID=1000
 # Install basic tools
 RUN yum install -y epel-release
 RUN yum install -y mock make rpm-build curl createrepo rpmlint redhat-lsb-core \
-                   git python-srpm-macros
+                   git python-srpm-macros dnf
```

Comment:
Should first submit a PR to packaging repository for this.
We can land that quick so that we can clear the diagnostic for this PR.

Reply:
@JohnMalmberg ok, added a PR here: https://github.com/daos-stack/packaging/pull/51

Reply:
Should we also update the Makefile_packaging.mk file to install with dnf?

Reply:
The CentOS dockerfile is not used for packaging.  The only thing that it is may be used for is testing the packages after they are built.   I would need to lookup how that test is run.

The Jenkinsfile for packaging is actually a groovy script in the pipeline-lib repository.

Reply:
The CentOS docker file in packaging is not currently used and will be removed in a future PR.

Reply:
@JohnMalmberg ok, but I had to manually update the Makefile_packaging.mk to use dnf or the build fails and the packaging check in jenkins goes back to yellow. Is this ok?

Reply:
I was mistaken, the CentOS 7 dockerfile is used for testing.  I missed it on my previous search.

While we use dnf because it is more efficient, our customers are going to use yum.

What is the failure with yum in the test?

Reply:
Found the log entry for run 20.   That looks like a repository bug.

Reply:
@JohnMalmberg I am not sure why but i keep getting this error in the build now:

[2021-07-16T18:22:36.030Z] ERROR: Exception(_topdir/SRPMS/mpifileutils-0.11-7.g8a22a2d.fc34.src.rpm) Config(epel-8-x86_64) 2 minutes 34 seconds

[2021-07-16T18:22:36.030Z] INFO: Results and/or logs in: /var/lib/mock/epel-8-x86_64/result

[2021-07-16T18:22:36.030Z] ERROR: Command failed: 

[2021-07-16T18:22:36.030Z]  # bash --login -c /usr/bin/rpmbuild -bb --target x86_64 --nodeps /builddir/build/SPECS/mpifileutils.spec

[2021-07-16T18:22:36.030Z] 

[2021-07-16T18:22:36.030Z] make: *** [packaging/Makefile_packaging.mk:338: chrootbuild] Error 1

Is there a way i can fix this?

Reply:
I can not reproduce the repository bug in a VM currently, can you revert it back to using yum once you get the rest of the builds fixed?

Reply:
@JohnMalmberg the build log actually reports this error:
[ 41%] Linking C executable dbcast
../common/libmfu.so.3.0.0: undefined reference to `daos_cont_open_by_label'
collect2: error: ld returned 1 exit status
make[2]: *** [src/dbcast/dbcast] Error 1
make[1]: *** [src/dbcast/CMakeFiles/dbcast.dir/all] Error 2
make: *** [all] Error 2
error: Bad exit status from /var/tmp/rpm-tmp.8ulsbk (%build)
RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.8ulsbk (%build)
Child return code was: 1
EXCEPTION: [Error()]

Because the newer mpifileutils uses the daos_cont_open_by_label which should exist in newer versions of DAOS?  But the odd thing is that *sometimes* it builds. It was previously, so I am not sure what is going on.

Reply:
In the install_pkgs.log file, you can find all the packages that were used for the build: 
daos-client-1.3.103-3.6721.g9d751014.el7.x86_64 1626447230 63595391 d0e2134ca2dcce7377af8871f434e25c installed
That is the build of daos that was used.

I can not find a reference to daos_cont_oopen_by_label in a checkout of the current master of daos.
Closest that I can find is a reference to dfuse_cont_open_by_label.

Reply:
> Should we also update the Makefile_packaging.mk file to install with dnf?

Yes, at a later date when someone updates the packaging for mpifileutils-pkg with the https://github.com/daos-stack/packaging/blob/master/install script this change would be overridden w/o this update. 

> What is the failure with yum in the test?

Yum fails to resolve the dependency for libcart.so.4 which is provided for by multiple packages in https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fmpifileutils-pkg/detail/PR-17/14/pipeline

As stated in https://github.com/daos-stack/mpifileutils-pkg/pull/17#issuecomment-877272035, switching to dnf resolves this dependency.